### DISCUSSION TITLE
Problems with ADT after updating to Air sdk 50.0

### DISCUSSION THREAD

**Discussion by OceanBreezeGames** [airsdk/Adobe-Runtime-Support#2343] [General]
I'm having troubles after updating to Harman air sdk 50.0. I downloaded manually and set it up as I have always done in the past. I'm not using the air package manager.

I've been using this command to package up my APK for years:
`%AIR_SDK%\bin\adt -package -target apk-captive-runtime -arch armv8  -storetype pkcs12 -keystore %KEYSTORE_PATH% -storepass password %GAME_NAME%_64bit.apk game-app.xml config.xml game.swf ./assets -C includes . -extdir d:/extensions/used
`

Now I get this error when I run it:

```
java.io.IOException: Unable to run java: com.adobe.air.ADTException: gradle tool failed:
FAILURE: Build failed with an exception.

* Where:
Settings file 'C:\Users\conta\AppData\Local\Temp\53fc1ccd-e919-43b8-881d-0f0d15475e79\settings.gradle'

* What went wrong:
Could not compile settings file 'C:\Users\conta\AppData\Local\Temp\53fc1ccd-e919-43b8-881d-0f0d15475e79\settings.gradle'.
> startup failed:
  General error during conversion: Unsupported class file major version 63

  java.lang.IllegalArgumentException: Unsupported class file major version 63
        at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
```

I've got Android Studio 4.1.2 installed, I see this in the Event log when opening:
```
JDK is configured
"19" is set to java version "19.0.1"
C:\Program Files\Java\jdk-19
```

But I also see this a little further down in the event log
```
Android Studio is using the following JDK location when running Gradle:
C:\Program Files\Android\Android Studio\jre
```

The Java version in that location (C:\Program Files\Android\Android Studio\jre) is: 1.8.0_242

My JAVA_HOME is set to C:\Program Files\Java\jdk-19

I feel like the problem is that ADT is trying to use the Android Studio JDK version for gradle builds which is 1.8.
I've tried setting a different Java SDK version in Android studio but it just doesn't take or it reverts back.

How can I force ADT to use the JDK 19? If that's the problem, if not then what is the problem?
This is something that I rarely touch and I don't really want to do a whole new setup for it.

Thanks!

**Comment by ajwfrost** [comment]
I'm wondering if you have another JDK somewhere that is being used by the ADT process itself... as the searching within ADT goes:
1) look in the `adt.cfg` file within `~/.airsdk/` (so on Windows, `c:\users\username\.airsdk\adt.cfg`)
2) look in the `adt.cfg` file within the AIR SDK "lib" folder
3) see what Java environment we're running in (and if it's JDK 11, this would be counted as sufficient..)
4) look for the JAVA_HOME environment variable
5) look for a jre folder within the normal Android Studio installation path

So hopefully you'd be able to fix this by creating/editing an `adt.cfg` file and adding in the JAVA_HOME definition there too.. (note, on Windows in this sort of property file, you'd need to use forward slashes, so `JAVA_HOME=c:/Program Files/Java/jdk-19`)

thanks

**Reply by OceanBreezeGames** [reply]
Thanks for the suggestions. I was able to get the adt.cfg setup. While searching the discussions for adt.cfg options I found this `<BuildLegacyAPK>true</BuildLegacyAPK>` which allowed me to build the APK file successfully and run it properly on my device so I think I'll stick with this method for now. 

I decided to test a non legacy mode build just to see if it would work. I enabled DebugOut in the adt.cfg file as well as setting the JAVA_HOME in there. It appears as though Gradle is using JRE 19, so I'm not really sure what the issue is. Partial output of the log is below, if you want the full one let me know, for now I"m happy with the legacy mode build.

```
*** ADT Logging at 2022/12/07 12:54:24 ***
Called by: D:\dev\air_sdk_50.0.1.3\lib\adt.jar
ADT 50.0.1.2 called with: -package -target apk-captive-runtime -arch armv8 -storetype pkcs12 -keystore d:\sdk\certificates\OBGProduction_11_20_2015.p12 -storepass xxxxxx christmassolitaire_64bit.apk game-app.xml config.xml game.swf ./assets -C includes . -extdir d:/extensions/used
Target requested was apk-captive-runtime, but not using legacy APK so building via Android Studio
Version in APK = 50.0.1.2, version of ADT = 50.0.1.2
Version in APK = 50.0.1.2, version of ADT = 50.0.1.2
Version in APK = 50.0.1.2, version of ADT = 50.0.1.2
Version in APK = 50.0.1.2, version of ADT = 50.0.1.2
Gradle build where ADT is using JRE version 19.0.1
but where Gradle will be using the JDK from c:/Program Files/Java/jdk-19\bin\java.exe
Gradle build command:  c:/Program Files/Java/jdk-19\bin\java.exe -Xmx1024m -Dorg.gradle.appname=gradlew -Dorg.gradle.jvmargs=-Xmx1024m -Duser.dir="C:\Users\conta\AppData\Local\Temp\aaede7a8-a2e3-48e4-83bd-feca0a9bf320" -cp "C:\Users\conta\AppData\Local\Temp\aaede7a8-a2e3-48e4-83bd-feca0a9bf320\gradle\wrapper\gradle-wrapper.jar";"D:\dev\air_sdk_50.0.1.3\lib\android\bin\gradle-cli.jar" org.gradle.wrapper.GradleWrapperMain :app:assembleRelease
..java.base/java.lang.Thread.getStackTrace(Thread.java:2550)
..com.adobe.air.DebugFile.printStack(DebugFile.java)
..com.adobe.air.ADTException.<init>(ADTException.java)
..com.adobe.air.apk.AABOutputStream.buildGradle(AABOutputStream.java)
..com.adobe.air.apk.AABPackager.renameOutput(AABPackager.java)
..com.adobe.air.ApplicationPackager.createPackage(ApplicationPackager.java)
..com.adobe.air.apk.AABPackager.createPackage(AABPackager.java)
..com.adobe.air.ADT.parseArgsAndGo(ADT.java)
..com.adobe.air.ADT.run(ADT.java)
..com.adobe.air.ADT.main(ADT.java)
Stack trace:
Unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: 
FAILURE: Build failed with an exception.

* Where:
Settings file 'C:\Users\conta\AppData\Local\Temp\aaede7a8-a2e3-48e4-83bd-feca0a9bf320\settings.gradle'

* What went wrong:
Could not compile settings file 'C:\Users\conta\AppData\Local\Temp\aaede7a8-a2e3-48e4-83bd-feca0a9bf320\settings.gradle'.
> startup failed:
  General error during conversion: Unsupported class file major version 63
```

**Reply by ajwfrost** [reply]
Interesting that the legacy mechanism worked - that's less likely as time goes by because it uses the older 'aapt' tool rather than 'aapt2'... if you are able to get gradle builds working, it would probably be a bit more future-proof (it's what Android Studio uses, the idea was that we'd be better hooking into their tools than trying to manually craft the APK or AAB files..)

I'm wondering whether the error is that there's some Java compilation going on using the JDK (where you're using JDK 19) and subsequently there's a tool that's trying to process that and failing because it's too recent.. so is it worth finding an installation of JDK 11 or similar, or I think we've seen that JDK !6 worked too, and then setting the JAVA_HOME value in the `adt.cfg` file so that it uses this?

thanks

**Reply by OceanBreezeGames** [reply]
OK, I tried JDK 11, removed the legacy tag from the xml file and it worked, apk created and ran fine on my device.
The adt.log confirmed using JDK 11 non-legacy mode.

Thanks!

**Reply by ajwfrost** [reply]
Great, thanks for confirming that!
